File: QuickBuy.Dominio/Entidades/Produto.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace QuickBuy.Dominio.Entidades
{
    public class Produto : Entidade
    {
        public int Id { get; set; }
        public string Nome { get; set; }
        public string Descricao { get; set; }
        public decimal Preco { get; set; }
        public string NomeArquivo { get; set; }

        public override void Validate()
        {
            try
            {
                if (Nome.Length == 0) AdicionarMensagemValidacao("Produto não possui nome.");
                if (Preco <= 0) AdicionarMensagemValidacao("Informar um preço válido.");
            }
            catch (NullReferenceException) {
                AdicionarMensagemValidacao("Produto não possui nome.");
            }

        }
    }
}

File: frontend/src/app/models/Produto.ts
export class Produto {
    id: number = 0;
    nome: string = '';
    descricao: string = '';
    preco: number = 0;
    nomeArquivo: string = '';
    precoOriginal: number = 0;
    quantidade: number = 0;

    constructor(data?: Partial<Produto>) {
        Object.assign(this, data);
    }
}
